src/crimson/osd/replicated_recovery_backend.cc

```diff
@@ -1003,13 +1011,21 @@ ReplicatedRecoveryBackend::handle_push(
   epoch_t epoch_frozen = pg.get_osdmap_epoch();
   DEBUGDPP("submitting transaction", pg);
 
-  co_await interruptor::make_interruptible(
-    shard_services.get_store().do_transaction(coll, std::move(t)));
-
   if (complete) {
+    auto ptiter = replica_push_targets.find(push_op.recovery_info.soid);
+    ceph_assert(ptiter != replica_push_targets.end());
+    auto manager = pg.obc_loader.get_obc_manager(ptiter->second);
+    manager.lock_excl_sync(); /* cannot already be locked */
```

Comment:
I'm missing something, we want to block reads on this push target obc between when we mark the
object not missing and when the commit completes.
Why is the lock taken here? Shouldn't we take it instead when we mark the object as not missing (`on_local_recover`)?

Reply:
nit: Fixes https://tracker.ceph.com/issues/69702

Reply:
on_local_recover gets called before that commit happens.